marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/ingestion/bulkInputCallerImpl.sjs

```diff
@@ -0,0 +1,27 @@
+'use strict';
+var endpointState; // jsonDocument?
+
+var input;         // jsonDocument*
+declareUpdate();
+
+const state  = fn.head(xdmp.fromJSON(endpointState));
+
+const work = fn.head(xdmp.fromJSON(workUnit));
+
+const inputs =
+    (input instanceof Sequence) ? input.toArray().map(item => fn.head(xdmp.fromJSON(item))) :
+    (input instanceof Document) ? [fn.head(xdmp.fromJSON(input))] :
+                                  [ {UNKNOWN: input} ];
+inputs.forEach(record => {
```

Comment:
Few comments on the for loop:

- Please indent the contents of the closure
- For permissions, use data-hub-common for read and update (we'll make these configurable later)
- Remove the console.log